File: typescript-examples/browser-sdk-showcase/api/go-to-url.ts
// https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/helpers/functions/goToUrl
import { BrowserContext, Page } from "playwright";
import { goToUrl } from "@intuned/browser";

interface Params {
  timeout?: number;
  retries?: number;
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  const timeout = params.timeout || 15000;
  const retries = params.retries || 3;

  // Navigate to the URL with enhanced reliability and automatic retries.
  // This will wait for the networkidle state more reliably than the default playwright behavior.
  // When waitForLoadingStateUsingAi=true, the function uses isPageLoaded under the hood to determine
  await goToUrl({
    page,
    url: "https://books.toscrape.com",
    timeoutInMs: timeout,
    retries,
    waitForLoadingStateUsingAi: true,
    // Since we are not passing the apiKey, the function will use the Intuned AI Gateway.
  });

  // Start your automation here, the page is already loaded and ready to use.
  return "Success";
}

File: python-examples/browser-sdk-showcase/api/go-to-url.py
# https://docs.intunedhq.com/automation-sdks/intuned-sdk/python/helpers/functions/go_to_url
from typing import TypedDict

from intuned_browser import go_to_url
from playwright.async_api import Page


class Params(TypedDict):
    timeout: int
    retries: int


async def automation(page: Page, params: Params, **_kwargs):
    timeout = params.get("timeout", 15)
    retries = params.get("retries", 3)

    # Navigate to the URL with enhanced reliability and automatic retries.
    # This will wait for the networkidle state more reliably than the default playwright behavior.
    # When wait_for_load_using_ai=True, the function uses is_page_loaded under the hood to determine
    await go_to_url(
        page,
        url="https://books.toscrape.com",
        timeout_s=timeout,
        retries=retries,
        wait_for_load_using_ai=True,
        # Since we are not passing the api_key, the function will use the Intuned AI Gateway.
    )

    # Start your automation here, the page is already loaded and ready to use.
    title = await page.title()
    url = page.url

    return {
        "title": title,
        "url": url,
    }
